tfhe/src/core_crypto/gpu/entities/lwe_packing_keyswitch_key.rs

```diff
@@ -0,0 +1,84 @@
+use crate::core_crypto::gpu::vec::CudaVec;
+use crate::core_crypto::gpu::{convert_lwe_keyswitch_key_async, CudaStreams};
+use crate::core_crypto::prelude::{
+    lwe_packing_keyswitch_key_input_key_element_encrypted_size, CiphertextModulus,
+    DecompositionBaseLog, DecompositionLevelCount, GlweSize, LweDimension,
+    LwePackingKeyswitchKeyOwned, PolynomialSize, UnsignedInteger,
+};
+
+#[derive(Debug)]
+pub struct CudaLwePackingKeyswitchKey<T: UnsignedInteger> {
+    pub(crate) d_vec: CudaVec<T>,
+    decomp_base_log: DecompositionBaseLog,
+    decomp_level_count: DecompositionLevelCount,
+    output_glwe_size: GlweSize,
+    output_polynomial_size: PolynomialSize,
+    ciphertext_modulus: CiphertextModulus<T>,
+}
+
+impl<T: UnsignedInteger> CudaLwePackingKeyswitchKey<T> {
+    pub fn from_lwe_packing_keyswitch_key(
+        h_ksk: &LwePackingKeyswitchKeyOwned<T>,
```

Comment:
Not sure if this is the right way to do it, @tmontaigu you probably know better.

Reply:
which right way are you looking for here ? There is some code already that does conversion/copies to the GPU right ?

Reply:
@pdroalves ?